src/common/Utilities.py

```diff
@@ -463,3 +463,38 @@ def str_to_int_list(str_list):
             int_item = list(range(strt, stop + 1))
         int_list = int_list + int_item
     return int_list
+
+def is_operator_adjoint(operator, num_tests = 5, max_err = 10e-5, verbose = False):
+    '''
+    Test if a given operator is adjoint.
+    The operator needs to have been already set_up() with valid objects.
+    The operator needs to have methods direct() and adjoint() implemented
+
+    Parameters
+    ----------
+    operator  :
+        Any SIRF operator that implements direct() and adjoint()
+    num_tests : int, optional
+        Square root of the number of tests with random data that will be executed. Default 5
+    max_err   : double, optional
+        Maximum allowed normalized error, tolerance. Change not recommended. Default 10e-5
+    verbose   : bool
+        Verbose option
+    '''
+    for iter1 in range(num_tests):
+        ## generate random data for x and direct()
+        x = operator.domain_geometry().allocate(value = 'random')
+        y_hat = operator.direct(x)
+        for iter2 in range(num_tests):
+            ## generate random data and adjoint()
+            y = operator.range_geometry().allocate( value = 'random')
+            x_hat = operator.adjoint(y)
+            # Check dot product identity
+            norm_err = abs(y_hat.dot(y) - x_hat.dot(x))/(abs(y_hat.dot(y))*0.5 + abs(x_hat.dot(x))*0.5)            
+            if norm_err > max_err:
+                if verbose:
+                    print(type(operator).__name__ + " is not adjoint, with normalized error of " + str(norm_err))
+                return False
+            elif verbose:
+                print("Testing " + type(operator).__name__ + ": Iteration " + str(iter1*num_tests+iter2+1) + "/" + str(num_tests**2))
+    return True
```

Comment:
should just be `return norm_err > max_err` and remove `return False` and `return True`.

Reply:
Also the verbose output for successful is a little misleading. To me, `Testing blah blah` implies that it's about to test that thing. More useful would be something along the lines of `Successfully completed iteration 1/25, with a norm of...`.

Reply:
But then I need to remove the `elif verbose`. Or you meant replace the False and the True by that condition?

Reply:
> But then I need to remove the elif verbose. Or you meant replace the False and the True by that condition?

Ah, sorry I understand now. You want to return False as soon as there's an error, else you keep going all the way to the end and then return `True`. Looks fine as is!

Reply:
@rijobro point taken with the verbose! I will work on that to print more useful info. This is still all WIP, so please keep bringing those suggestions ;)